### PR TITLE
pump client: write commit binlog will never return error

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
write commit binlog should not return error to tidb.

### What is changed and how it works?
just return nil if retry more than 10 minutes.
and add unit for it.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@july2993 @GregoryIan PTAL

File: tidb-binlog/pump_client/client.go [hunk 1]
```diff
@@ -251,22 +254,23 @@ func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
...
+				err = errors.New(resp.Errmsg)
+			}
+			if err == nil {
+				return nil
+			}
+			Logger.Errorf("[pumps client] write binlog error %v", err)
 		}
-		Logger.Errorf("[pumps client] write binlog error %v", err)
 
 		if binlog.Tp == pb.BinlogType_Commit {
```
Comment (WangXiangUSTC) [inline tidb-binlog/pump_client/client.go]: do we need return error for rollback binlog?

**Comment by IANTHEREAL** [comment]
@july2993 PTAL

File: tidb-binlog/pump_client/client.go (hunk 1 above)
Comment (july2993) [reply tidb-binlog/pump_client/client.go]: rollback should be same as comit

File: tidb-binlog/pump_client/client.go [hunk 2]
```diff
@@ -295,6 +299,10 @@ func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
 
 // setPumpAvaliable set pump's isAvaliable, and modify UnAvaliablePumps or AvaliablePumps.
 func (c *PumpsClient) setPumpAvaliable(pump *PumpStatus, avaliable bool) {
+	if pump == nil {
```
Comment (july2993) [inline tidb-binlog/pump_client/client.go]: you should better handle abnormality case ASAP

File: tidb-binlog/pump_client/client.go [hunk 3]
```diff
@@ -251,22 +254,23 @@ func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
...
-			return ErrNoAvaliablePump
-		}
-
-		resp, err = pump.writeBinlog(req, c.BinlogWriteTimeout)
-		if err == nil && resp.Errmsg != "" {
-			err = errors.New(resp.Errmsg)
-		}
-		if err == nil {
-			return nil
+			err = ErrNoAvaliablePump
```
Comment (IANTHEREAL) [inline tidb-binlog/pump_client/client.go]: when to return nil pump from selector？

Comment (IANTHEREAL) [reply tidb-binlog/pump_client/client.go]: same logic at L241

Comment (WangXiangUSTC) [reply tidb-binlog/pump_client/client.go]: when no pump in selector.

File: tidb-binlog/pump_client/client_test.go [hunk 4]
```diff
@@ -190,8 +190,30 @@ func (t *testClientSuite) TestWriteBinlog(c *C) {
...
+			PrewriteValue: make([]byte, 1),
+		}
+		commitBinlog := &pb.Binlog{
+			Tp:            pb.BinlogType_Commit,
+			StartTs:       123,
+			CommitTs:      123,
+			PrewriteValue: make([]byte, 1),
+		}
+
+		err = pumpClient.WriteBinlog(preWriteBinlog)
```
Comment (IANTHEREAL) [inline tidb-binlog/pump_client/client_test.go:206]: ci was failed

**Comment by IANTHEREAL** [comment]
LGTM

**Review by july2993 — APPROVED** [review]
LGTM

**Pull request merged**